modules/common/src/main/scala/surge/internal/utils/ExecutionContextProber.scala

```diff
@@ -0,0 +1,155 @@
+// Copyright © 2017-2021 UKG Inc. <https://www.ukg.com>
+
+package surge.internal.utils
+
+import akka.Done
+import akka.actor._
+import akka.pattern.pipe
+import surge.internal.utils.ExecutionContextProberActor.Messages.Stop
+import surge.internal.utils.ExecutionContextProberActor.TimerKeys.{ SendProbesKey, TimeoutKey }
+
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.{ FiniteDuration, _ }
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.language.postfixOps
+
+final case class ExecutionContextProberSettings(
+    targetEc: ExecutionContext,
+    initialDelay: FiniteDuration,
+    timeout: FiniteDuration,
+    interval: FiniteDuration,
+    numProbes: Int)
+
+object ExecutionContextProberActor {
+
+  def props(settings: ExecutionContextProberSettings): Props = Props(new ExecutionContextProberActor(settings))
+
+  object Messages {
+
+    final case object Stop
+
+    final case object SendProbes
+
+    final case object Timeout
+
+    final case object HasIssues // used by the unit test
+
+  }
+
+  object TimerKeys {
+
+    final case object SendProbesKey
+
+    final case object TimeoutKey
+
+  }
+
+  sealed trait ProbingResult
+
+  final case class Ack(id: UUID) extends ProbingResult
+
+  def noOpFuture(id: UUID)(implicit ec: ExecutionContext): Future[ProbingResult] = Future {
+    Ack(id)
+  }
+
+  val warningText = "Possible thread starvation issue"
+
+}
+
+class ExecutionContextProberActor(settings: ExecutionContextProberSettings) extends Actor with ActorLogging with Timers {
+
+  import ExecutionContextProberActor.Messages._
+  import ExecutionContextProberActor._
+
+  implicit val system: ActorSystem = context.system
+
+  var detectedIssue = false // used by the unit test
```

Comment:
I'd keep this private since it's pretty internal to the actor

Reply:
Yes, so this is interesting, because in idiomatic Akka "private" doesn't make much sense, since all internal state is private anyway. So, in other words:

```
val myActor: ActorRef = system.actorOf(Props(new MyActor()), name = "myActor") 
myActor.detectedIssues // this won't work anyway - it doesn't make a difference if it's private or public
```

Reply:
@garyKeorkunian Fixed (now the variable is inside the receive context) ;-)